t/plugin/authz-casdoor.t

```diff
@@ -0,0 +1,444 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+use t::APISIX 'no_plan';
+
+repeat_each(1);
+no_long_string();
+no_root_location();
+add_block_preprocessor(sub {
+    my ($block) = @_;
+
+    if ((!defined $block->error_log) && (!defined $block->no_error_log)) {
+        $block->set_value("no_error_log", "[error]");
+    }
+
+    if (!defined $block->request) {
+        $block->set_value("request", "GET /t");
+    }
+
+    my $http_config = $block->http_config // <<_EOC_;
+    server {
+        listen 10420;
+        location /api/login/oauth/access_token {
+            content_by_lua_block {
+                local json_encode = require("toolkit.json").encode
+                ngx.req.read_body()
+                local arg = ngx.req.get_post_args()["code"]
+
+                local core = require("apisix.core")
+                local log = core.log
+
+                if arg == "wrong" then
+                    ngx.status = 200
+                    ngx.say(json_encode({ access_token = "bbbbbbbbbb", expires_in = 0 }))
+                    return
+                end
+
+                ngx.status = 200
+                ngx.say(json_encode({ access_token = "aaaaaaaaaaaaaaaa", expires_in = 1000000 }))
+            }
+        }
+    }
+_EOC_
+
+    $block->set_value("http_config", $http_config);
+});
+run_tests();
+
+__DATA__
+
+=== TEST 1: sanity
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.authz-casdoor")
+            local fake_uri = "http://127.0.0.1:" .. ngx.var.server_port
+            local callback_url = "http://127.0.0.1:" .. ngx.var.server_port ..
+                                    "/anything/callback"
+            local conf = {
+                callback_url = callback_url,
+                endpoint_addr = fake_uri,
+                client_id = "7ceb9b7fda4a9061ec1c",
+                client_secret = "3416238e1edf915eac08b8fe345b2b95cdba7e04"
+            }
+            local ok, err = plugin.check_schema(conf)
+            if not ok then
+                ngx.say(err)
+            end
+
+            local conf2 = {
+                callback_url = callback_url .. "/?code=aaa",
+                endpoint_addr = fake_uri,
+                client_id = "7ceb9b7fda4a9061ec1c",
+                client_secret = "3416238e1edf915eac08b8fe345b2b95cdba7e04"
+            }
+            ok, err = plugin.check_schema(conf2)
+            if ok then
+                ngx.say("err")
+            end
+
+            local conf3 = {
+                callback_url = callback_url,
+                endpoint_addr = fake_uri .. "/",
+                client_id = "7ceb9b7fda4a9061ec1c",
+                client_secret = "3416238e1edf915eac08b8fe345b2b95cdba7e04"
+            }
+            ok, err = plugin.check_schema(conf3)
+            if ok then
+                ngx.say("err")
```

Comment:
```suggestion
                ngx.say(err)
```

Reply:
if ok==true or err here is empty, then this test should fail, so I think it's right to use "err" rather than err

Reply:
I have made revisions to make the  information more clear